src/concrete_convex_hull.jl

```diff
@@ -136,6 +119,48 @@ function convex_hull!(points::Vector{VN};
     end
 end
 
+function _two_points_2d!(points)
+
+    n = length(first(points))
+
+    if n == 1
+        p1, p2 = points[1], points[2]
+        if p1 == p2  # check for redundancy
+            pop!(points)
+        elseif p1[1] > p2[1]
+            points[1], points[2] = p2, p1
+        end
+    elseif n == 2
+        # special case, see #876
+        p1, p2 = points[1], points[2]
+        if p1 == p2  # check for redundancy
+            pop!(points)
+        elseif p1 <= p2
+            nothing
+        else
+            points[1], points[2] = p2, p1
+        end
+    end
+    return points
+end
+
+function _three_points_2d!(points)
+    A, B, C = points[1], points[2], points[3]
+    turn = (A[2] - B[2]) * C[1] + (B[1] - A[1]) * C[2] + (A[1] * B[2] - B[1] * A[2])
+
+    if isapproxzero(turn)
+        # ABC are collinear
+        points[1], points[2] = points[argmin([A[1], B[1], C[1]])], points[argmax([A[1], B[1], C[1]])]
+        pop!(points)
+        _two_points_2d!(points)
+    elseif turn < 0
```

Comment:
```suggestion
    elseif turn < zero(N)
```
For this to work, you need to annotate the type by writing `points::AbstractVector{<:AbstractVector{N}}) where {N<:Real}`.

Reply:
You still need to add this type information.